domain/machine/state/machine_cloud_instance.go

```diff
@@ -254,6 +256,45 @@ WHERE  machine_uuid = $machineUUID.uuid;`
 	return instanceId, nil
 }
 
+// InstanceName returns the cloud specific instance display name for this
+// machine.
+// [machineerrors.NotProvisionedError].
+func (st *State) InstanceName(ctx context.Context, mUUID string) (string, error) {
+	db, err := st.DB()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	mUUIDParam := machineUUID{UUID: mUUID}
+	query := `
+SELECT &instanceData.display_name
+FROM   machine_cloud_instance
+WHERE  machine_uuid = $machineUUID.uuid;`
+	queryStmt, err := st.Prepare(query, mUUIDParam, instanceData{})
```

Comment:
Custom type here instead of the full-fat `instanceData`

Reply:
Sure, done!